pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -577,35 +588,42 @@ func (c *Compactor) runCompact(ctx context.Context, logger log.Logger, job Job,
 	var bloomBlocksRefs []bloomshipper.BlockRef
 	var tombstonedBlockRefs []bloomshipper.BlockRef
 
+	fpRate := c.limits.BloomFalsePositiveRate(job.Tenant())
 	metas, err := bloomShipperClient.GetMetas(ctx, metaSearchParams)
 	if err != nil {
 		return err
 	}
 
 	if len(metas) == 0 {
 		// Get chunks data from list of chunkRefs
-		chks, err := storeClient.chunk.GetChunks(ctx, makeChunkRefs(job.Chunks(), job.Tenant(), job.Fingerprint()))
-		if err != nil {
-			return err
-		}
 
-		fpRate := c.limits.BloomFalsePositiveRate(job.Tenant())
-		storedBlocks, err := CompactNewChunks(ctx, logger, job, chks, bt, fpRate, bloomShipperClient, c.cfg.WorkingDirectory)
-		if err != nil {
-			return level.Error(logger).Log("compacting new chunks", err)
-		}
+		for _, index := range job.indices {
+			chks, err := storeClient.chunk.GetChunks(ctx, makeChunkRefs(index.Chunks(), job.Tenant(), index.Fingerprint()))
```

Comment:
it looks like this line will download all the chunks to memory ... we need to use lazyChunk to download the chunk, immediately process it, and allow GC to release the memory. wdyt?

Reply:
Good point, totally agreed. But AFAICT [lazyChunk](https://github.com/grafana/loki/blob/main/pkg/storage/store.go#L384-L421) is not exposed (please let me know if i'm looking at the wrong place). Making a change will result in a larger PR. Should we address that later?